tests/test_iib_build_details_model.py

```diff
@@ -1,72 +1,394 @@
-        "omps_operator_version": {"operator": "1.0"},
+        "state": "in_progress",
+        "state_reason": "state_reason",
+        "request_type": "regenerate-bundle",
+        "state_history": [],
+        "batch": 1,
+        "batch_annotations": {"batch_annotations": 1},
+        "logs": {},
+        "updated": "updated",
+        "user": "user@example.com",
+        "bundle_image": "bundle_image",
+        "from_bundle_image": "from_bundle_image",
+        "from_bundle_image_resolved": "from_bundle_image_resolved",
+        "organization": "organization",
+    }
+    return json
+
+
+@pytest.fixture
+def fixture_unknown_build_details_json():
+    json = {
+        "id": 3,
+        "arches": ["x86_64"],
+        "state": "in_progress",
+        "state_reason": "state_reason",
+        "request_type": "unknown",
+        "state_history": [],
+        "batch": 1,
+        "batch_annotations": {"batch_annotations": 1},
+        "logs": {},
+        "updated": "updated",
+        "user": "user@example.com",
+        "bundle_image": "bundle_image",
+        "from_bundle_image": "from_bundle_image",
+        "from_bundle_image_resolved": "from_bundle_image_resolved",
+        "organization": "organization",
     }
     return json
 
 
-def test_iibbuilddetailsmodel(fixture_build_details_json):
-    unexpected_model = IIBBuildDetailsModel(
-        1,
-        "finished",
-        "state_reason",
-        [],
-        "from_index",
-        "from_index_resolved",
-        ["bundles1"],
-        ["operator1"],
-        "organization",
-        "binary_image",
-        "binary_image_resolved",
-        "index_image",
-        "request_type",
-        ["x86_64"],
-        {"bundle_mapping": "map"},
-        {"operator": "1.0"},
+@pytest.fixture
+def fixture_bundle_image_missing_json():
+    json = {
+        "id": 3,
+        "arches": ["x86_64"],
+        "state": "in_progress",
+        "state_reason": "state_reason",
+        "request_type": "regenerate-bundle",
+        "state_history": [],
+        "batch": 1,
+        "batch_annotations": {"batch_annotations": 1},
+        "logs": {},
+        "updated": "updated",
+        "user": "user@example.com",
+        "from_bundle_image": "from_bundle_image",
+        "from_bundle_image_resolved": "from_bundle_image_resolved",
+        "organization": "organization",
+    }
+    return json
+
+
+@pytest.fixture
+def fixture_optional_args_missing_json():
+    json = {
+        "id": 3,
+        "arches": ["x86_64"],
+        "state": "in_progress",
+        "state_reason": "state_reason",
+        "request_type": "regenerate-bundle",
+        "batch": 1,
+        "updated": "updated",
+        "user": "user@example.com",
+        "bundle_image": "bundle_image",
+        "from_bundle_image": "from_bundle_image",
+        "from_bundle_image_resolved": "from_bundle_image_resolved",
+        "organization": "organization",
+    }
+    return json
+
+
+def test_from_dict_success(
```

Comment:
Could not this be somehow made simpler when `pytests.mark.parametrize` is used?

Reply:
As we discussed, how the tests are made will be changed in another PR.